lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -2508,7 +2523,8 @@ class AvailabilityWalker : public ASTWalker {
         auto *decl = component.getDeclRef().getDecl();
         auto loc = component.getLoc();
         SourceRange range(loc, loc);
-        diagAvailability(decl, range, nullptr);
+        diagAvailability(decl, range, nullptr,
+                         DeclAvailabilityFlag::ForObjCKeyPath);
```

Comment:
Oops, this needs to check it's actually an ObjC rather than Swift keypath first. Fixing...